server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java

```diff
@@ -88,7 +88,7 @@ protected void newResponseAsync(
         if (NodesStatsRequestParameters.Metric.ALLOCATIONS.containedIn(metrics)) {
             client.execute(
                 TransportGetAllocationStatsAction.TYPE,
-                new TransportGetAllocationStatsAction.Request(),
+                new TransportGetAllocationStatsAction.Request(task.getParentTaskId()),
```

Comment:
This needs to be the ID of the task itself, not its parent. This is a child of the node stats task, not a sibling.